### PR TITLE
HS-951 Make Adapter functional

### PR DESCRIPTION
**Pull request by FrancescoBuc86**
https://opennms.atlassian.net/browse/HS-951

### CODE REVIEW

**Comment by FrancescoBuc86** [comment]
Tests to be added, on it right now

**Review by fooker — CHANGES_REQUESTED** [review]
Having adapters here does not make any sense. They just add another layer of complexity without adding any functionality at all.

File: minion/minion-grpc/grpc-features/src/main/feature/feature.xml [hunk 1]
```diff
@@ -38,21 +38,36 @@
...
-    </feature>
+           <bundle>mvn:org.opennms.horizon.shared/horizon-common-logging/${project.version}</bundle>
+           <bundle>mvn:org.opennms.horizon.shared/ipc-api/${project.version}</bundle>
+           <bundle>mvn:org.opennms.horizon.minion/grpc-client/${project.version}</bundle>
+           <bundle>mvn:com.fasterxml.jackson.core/jackson-databind/${jackson.version}</bundle>
+           <bundle>mvn:com.fasterxml.jackson.core/jackson-core/${jackson.version}</bundle>
+           <bundle>mvn:com.fasterxml.jackson.core/jackson-annotations/${jackson.version}</bundle>
+           <bundle>mvn:com.google.code.gson/gson/${gson.version}</bundle>
+           <bundle>wrap:mvn:com.google.protobuf/protobuf-java/${protobuf.version}</bundle>
+           <bundle>mvn:com.google.guava/guava/${guava.version}</bundle>
```
Comment (freddy61025) [inline minion/minion-grpc/grpc-features/src/main/feature/feature.xml:69]: it is already in internal-grpc-guava.

File: minion/minion-grpc/grpc-features/src/main/feature/feature.xml [hunk 2]
```diff
@@ -38,21 +38,36 @@
...
-        <bundle>mvn:org.opennms.horizon.minion/grpc-client/${project.version}</bundle>
-    </feature>
+           <bundle>mvn:org.opennms.horizon.shared/horizon-common-logging/${project.version}</bundle>
+           <bundle>mvn:org.opennms.horizon.shared/ipc-api/${project.version}</bundle>
+           <bundle>mvn:org.opennms.horizon.minion/grpc-client/${project.version}</bundle>
+           <bundle>mvn:com.fasterxml.jackson.core/jackson-databind/${jackson.version}</bundle>
+           <bundle>mvn:com.fasterxml.jackson.core/jackson-core/${jackson.version}</bundle>
+           <bundle>mvn:com.fasterxml.jackson.core/jackson-annotations/${jackson.version}</bundle>
+           <bundle>mvn:com.google.code.gson/gson/${gson.version}</bundle>
+           <bundle>wrap:mvn:com.google.protobuf/protobuf-java/${protobuf.version}</bundle>
```
Comment (freddy61025) [inline minion/minion-grpc/grpc-features/src/main/feature/feature.xml:68]: it is in grpc-protobuf, seems duplicated.

File: shared-lib/horizon-ipc/ipc-grpc/ipc-grpc-contract/pom.xml [hunk 3]
```diff
@@ -54,6 +54,11 @@
             <artifactId>proto</artifactId>
             <version>${project.version}</version>
         </dependency>
+        <dependency>
```
Comment (freddy61025) [inline shared-lib/horizon-ipc/ipc-grpc/ipc-grpc-contract/pom.xml:57]: why we need to modify shared-lib?

File: minion/minion-flows/flows-parser/src/main/java/org/opennms/horizon/minion/flows/adapter/common/AbstractFlowAdapter.java [hunk 4]
```diff
@@ -73,95 +81,73 @@
...
+        LOG.debug("Sending Packets and flows to metrics-processor for Enrichment step. ");
 
-    protected abstract List<Flow> convert(final P packet, final Instant receivedAt);
+        telemetryRegistry.getDispatcher().send(buildMessage(flowSourceProto, flows,
+            this.applicationThresholding, this.applicationDataCollection, this.packages));
 
-    public void destroy() {
-        // not needed
-    }
+        LOG.debug("Completed Adapter and Enrichment step for {} telemetry message.", telemetryMessage);
```
Comment (freddy61025) [inline minion/minion-flows/flows-parser/src/main/java/org/opennms/horizon/minion/flows/adapter/common/AbstractFlowAdapter.java:140]: Is this message accurate? Enrichment is not done at here right?

File: minion/minion-flows/flows-parser/src/main/java/org/opennms/horizon/minion/flows/adapter/common/AbstractFlowAdapter.java [hunk 5]
```diff
@@ -73,95 +81,73 @@
...
 
-    public String getMetaDataNodeLookup() {
-        return metaDataNodeLookup;
     }
 
+    protected abstract P parse(TelemetryMessage message);
```
Comment (freddy61025) [inline minion/minion-flows/flows-parser/src/main/java/org/opennms/horizon/minion/flows/adapter/common/AbstractFlowAdapter.java:144]: It is better to leave comment about what is that function about.

File: minion/minion-flows/flows-parser/src/main/java/org/opennms/horizon/minion/flows/adapter/common/AbstractFlowAdapter.java [hunk 6]
```diff
@@ -73,95 +81,73 @@
...
+
+    protected abstract List<Flow> convert(final P packet, final Instant receivedAt);
```
Comment (freddy61025) [inline minion/minion-flows/flows-parser/src/main/java/org/opennms/horizon/minion/flows/adapter/common/AbstractFlowAdapter.java:146]: It is better to leave comment about what is that function about.

**Comment by FrancescoBuc86** [comment]
> Having adapters here does not make any sense. They just add another layer of complexity without adding any functionality at all.

Hi Dustin, can you explain what do you mean by "here"?

**Review by freddy61025 — CHANGES_REQUESTED** [review]
Two more questions.
1. Have you tried to turn on tilt and monitor Kafka results?
2. Have you copied all adapter related test?

File: minion/minion-flows/flows-parser/src/main/java/org/opennms/horizon/minion/flows/parser/ParserBase.java [hunk 7]
```diff
@@ -28,6 +28,8 @@
 
 package org.opennms.horizon.minion.flows.parser;
 
+import static java.util.Objects.nonNull;
```
[the remark this answers is not in the document]
Comment (FrancescoBuc86) [reply minion/minion-flows/flows-parser/src/main/java/org/opennms/horizon/minion/flows/parser/ParserBase.java:31]: ok

File: minion/minion-flows/flows-parser/src/test/java/org/opennms/horizon/minion/flows/parser/AdapterHolderTest.java [hunk 8]
```diff
@@ -0,0 +1,84 @@
+package org.opennms.horizon.minion.flows.parser;
+
+import static org.junit.Assert.*;
```
[the remark this answers is not in the document]
Comment (FrancescoBuc86) [reply minion/minion-flows/flows-parser/src/test/java/org/opennms/horizon/minion/flows/parser/AdapterHolderTest.java:3]: ok

**Pull request closed without merging**